libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -897,6 +897,38 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			m_context << success;
 			break;
 		}
+		case FunctionType::Kind::ABIEncode:
+		case FunctionType::Kind::ABIEncodePacked:
+		{
+			TypePointers argumentTypes;
+			for (auto const& arg: arguments)
+			{
+				arg->accept(*this);
+				argumentTypes.push_back(arg->annotation().type);
+			}
+			utils().fetchFreeMemoryPointer();
+			// stack now: <memory pointer>
+
+			// adjust by 32 bytes to accommodate the length
+			m_context << u256(32) << Instruction::ADD;
+			utils().encodeToMemory(argumentTypes, TypePointers(), function.padArguments(), true);
```

Comment:
`copyDynamicDataInPlace` should only be true if `padArguments` is also true.

Reply:
This is a copy of the `SHA3` case above. The only difference is the `+32` for the offset.

Reply:
Yes and `sha3` does not use ABI encoding.